web/packages/teleport/src/components/hooks/useUrlFiltering/useUrlFiltering.ts

```diff
@@ -39,7 +40,8 @@ export interface UrlFilteringState {
 }
 
 export function useUrlFiltering(
-  initialParams: Partial<ResourceFilter>
+  initialParams: Partial<Omit<ResourceFilter, 'includedResourceMode '>>,
+  includedResourceMode?: IncludedResourceMode
```

Comment:
How does `includedResourceMode` gets updated? `useUrlFiltering` exposes `setParams` that accepts the `includedResourceMode` field, but as far as I understand it has no effect?

FWIW, passing an arg through a hook like that (you accept something as an input and then pass it to the output with no changes) indicates that this could've been done outside of the hook. But I don't know how difficult it'd be to refactor this.

Reply:
Updating the `availabilityFilter` prop passed to the unified resources component (`ClusterResources`) updates the `includedResourceMode` value passed to `useUrlFiltering`, and in `useUrlFiltering` when updating the `params`, we include that `includedResourceMode` in the returned object:

```
  const params = useMemo(() => {
    const urlParams = getResourceUrlQueryParams(search);
    return {
      ...initialParamsState,
      ...urlParams,
      includedResourceMode,
      pinnedOnly:
        urlParams.pinnedOnly !== undefined
          ? urlParams.pinnedOnly
          : initialParamsState.pinnedOnly,
    };
  }, [search, includedResourceMode]);
  ```